src/Integrations/Integrations/PDO/PDOIntegration.php

```diff
@@ -133,6 +133,20 @@ public static function detectError($pdoOrStatement, SpanData $span)
         }
         $errorInfo = $pdoOrStatement->errorInfo();
         $span->meta[Tag::ERROR_MSG] = 'SQL error: ' . $errorCode . '. Driver error: ' . $errorInfo[1];
+
+        // Driver-specific error message will be in the rest of the array
+        // The spec suggests the error message to be at pos 2 only,
+        // for ODBC: the SQLSTATE in [3] (but e.g. SQLServer doc states that the array can contain monre than 3 elements)
+        // for DBLIB: in [3] an OSerror (integer), in [4] a severity and optionally a string representation of the OSerror in [5].
+        if (count($errorInfo) > 2) {
+            $additionalInfo = [];
+            for ($i = 2; $i < count($errorInfo); $i++) {
+                $additionalInfo[] = $errorInfo[$i];
+            }
+
+            $span->meta[Tag::ERROR_MSG] .= '. Driver-specific error data: ' . implode('. ', $additionalInfo);
```

Comment:
```suggestion
            $span->meta[Tag::ERROR_MSG] .= '. Driver-specific error data: ' . implode('. ', array_slice($additionalInfo, 2));
```

Reply:
Introduced in last commit (didn't use the exact suggestion as we need to slice errorInfo)